tests/fs-test.rs

```diff
@@ -19,93 +24,236 @@ pub fn verify_file_equals(testfile: &PathBuf, contents: &str) {
 	assert_eq!(file_content, contents.to_string());
 }
 
+/// This checks whether a VM can create and write to a file on the host.
 #[test]
-fn new_file_test() {
+#[serial]
+fn create_mapped_parent_nonpresent_file() {
 	env_logger::try_init().ok();
+	let fixture_path = get_fs_fixture_path();
+	// Tests successful directory traversal starting from file in child
+	// directory of a mapped directory.
+	let guest_path = PathBuf::from("/root/");
+	let mut host_path = fixture_path;
+	host_path.push("this_folder_exists");
+	let mut file_path = host_path.clone();
+	file_path.push("foo.txt");
+	remove_file_if_exists(&file_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		guest_path.to_str().unwrap()
+	)]
+	.to_vec();
+
 	let params = Params {
 		cpu_count: 2.try_into().unwrap(),
 		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
 			.unwrap()
 			.try_into()
 			.unwrap(),
-		output: Output::Buffer,
-		stats: true,
+		file_mapping: uhyvefilemap_params.clone(),
+		#[cfg(target_os = "linux")]
+		file_isolation: FileSandboxMode::Strict,
 		..Default::default()
 	};
 
 	let bin_path = build_hermit_bin("open_close_file");
 	let vm = UhyveVm::new(bin_path, params).unwrap();
 	let res = vm.run(None);
+	remove_file_if_exists(&file_path);
 	check_result(&res);
 }
 
 #[test]
-fn uhyvefilemap_test() {
+#[serial]
+fn create_write_unmapped_nonpresent_file() {
 	env_logger::try_init().ok();
-	let output_path = PathBuf::from("foo.txt");
-	remove_file_if_exists(&output_path);
-	let bin_path = build_hermit_bin("create_file");
+	let fixture_path = get_fs_fixture_path();
+	let mut host_path = fixture_path;
+	host_path.push("file_to_write.txt");
+	remove_file_if_exists(&host_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		"/root/dir/wrong.txt"
+	)]
+	.to_vec();
 
-	let mut params = Params {
+	let params = Params {
 		cpu_count: 2.try_into().unwrap(),
 		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
 			.unwrap()
 			.try_into()
 			.unwrap(),
-		file_mapping: vec!["/root/foo.txt:wrong.txt".to_string()],
+		file_mapping: uhyvefilemap_params,
+		#[cfg(target_os = "linux")]
+		file_isolation: FileSandboxMode::Strict,
 		output: Output::Buffer,
 		stats: true,
 		..Default::default()
 	};
 
 	// The file should not exist on the host OS.
-	let mut vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
-	let mut res = vm.run(None);
+	let bin_path = build_hermit_bin("create_file");
+	let vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
+	let res = vm.run(None);
 	check_result(&res);
-	assert!(!output_path.exists());
+	assert!(!host_path.exists());
+	remove_file_if_exists(&host_path);
```

Comment:
Doesn't make sense with Landlock - we should create a new directory that is `.gitignore`'d instead.